gvr-immersivepedia/gvrimmersivepedia/src/main/java/org/gearvrf/immersivepedia/dinosaur/DinosaurFactory.java

```diff
@@ -88,53 +91,91 @@ private Dinosaur createDinosauros(String name, int dinoMeshId, int dinoTextureId
 
     private FocusableSceneObject createDinosaur(int dinoMeshId, int dinoTextureId) {
         GVRMesh baseMesh = gvrContext.loadMesh(new GVRAndroidResource(gvrContext, dinoMeshId), settings);
-        GVRTexture baseTexture = gvrContext.loadTexture(new GVRAndroidResource(gvrContext, dinoTextureId));
-        FocusableSceneObject dino = new FocusableSceneObject(gvrContext, baseMesh, baseTexture);
+        GVRTexture baseTexture = gvrContext.loadTexture(new GVRAndroidResource(gvrContext, R.drawable.empty));
+        final FocusableSceneObject dino = new FocusableSceneObject(gvrContext, baseMesh, baseTexture);
+        gvrContext.loadTexture(new TextureCallback() {
+
+            @Override
+            public void loaded(GVRTexture arg0, GVRAndroidResource arg1) {
+                dino.getRenderData().getMaterial().setMainTexture(arg0);
+            }
+
+            @Override
+            public void failed(Throwable arg0, GVRAndroidResource arg1) {
+                // TODO Auto-generated method stub
+            }
+
+            @Override
+            public boolean stillWanted(GVRAndroidResource arg0) {
+                // TODO Auto-generated method stub
+                return false;
+            }
+        }, new GVRAndroidResource(gvrContext, dinoTextureId));
+
         return dino;
     }
 
     private FocusableSceneObject createDinosaurBase(int baseMeshId) {
-
         GVRMesh baseMesh = gvrContext.loadMesh(new GVRAndroidResource(gvrContext, baseMeshId), settings);
-        GVRTexture baseTexture = gvrContext.loadTexture(new GVRAndroidResource(gvrContext, R.drawable.base_tex_diffuse));
+        GVRTexture baseTexture = gvrContext.loadTexture(new GVRAndroidResource(gvrContext, R.raw.base_tex_diffuse));
```

Comment:
This uses a version of loadTexture that expects a bitmap image as an input. You should be using an overload that explicitly mentions that it tries to detect the texture type. The app definitely crashes on my S7.